packages/salesforce-adapter/src/client/client.ts

```
@@ -564,19 +564,31 @@ export default class SalesforceClient {
     queryString: string,
     useToolingApi = false,
   ): AsyncIterable<SalesforceRecord[]> {
-    const hadMore = (results: QueryResult<Value>): boolean =>
-      !_.isUndefined(results.nextRecordsUrl)
+    const nextResultsPage = (results: ExtendedQueryResult): string | undefined => (
+      // we could use queryLocator for tooling and nextRecordsUrl for the other calls,
+      // but since currently queryLocator is not in the jsforce types, it seems safer
+      // to have it as a fallback in case the reply does end up getting populated in nextRecordsUrl
+      results.nextRecordsUrl ?? results.queryLocator
+    )
 
     const conn = useToolingApi ? this.conn.tooling : this.conn
     let results = await conn.query(queryString)
     yield results.records as SalesforceRecord[]
 
-    let hasMore = hadMore(results)
-    while (hasMore) {
-      // eslint-disable-next-line no-await-in-loop
-      results = await conn.queryMore(results.nextRecordsUrl as string)
-      yield results.records as SalesforceRecord[]
-      hasMore = hadMore(results)
+    let nextQueryLocator = nextResultsPage(results)
+    try {
+      while (nextQueryLocator !== undefined) {
+        log.debug('received next query locator for query \'%s\'%s: %s',
+          queryString, useToolingApi ? ' (tooling)' : '', nextQueryLocator)
+        // eslint-disable-next-line no-await-in-loop
+        results = await conn.queryMore(nextQueryLocator)
+        yield results.records as SalesforceRecord[]
+        nextQueryLocator = nextResultsPage(results)
+      }
+    } catch (e) {
+      // avoid failing the entire operation
```

we should not suppress errors in fetch, that could lead to inconsistent fetch results (we had a lot of trouble with this kind of thing in the past).
allowing the flow to continue with less information as if there was no error would lead to the next fetch potentially returning a different result which would be very visible (and confusing) to the user